firmware/robot2015/cpu/ball_sense.hpp

```diff
@@ -0,0 +1,38 @@
+// Determines if the emitter to reciever beam is broken
+//  while accounting for ambiant light
+#include "mbed.h"
+
+class ball_sense {
+public:
+    ball_sense();
+
+    // Call once per cycle
+    void update_ball_sensor();
+
+    // Returns true if beam is broken (Ball is contained)
+    bool have_ball();
+
+private:
+    DigitalOut* emitter_pin;
+    AnalogIn* detector_pin;
+
+    // Holds the light sensed when
+    //  emitter is lit (sense_light) and
+    //  dark (sense_dark)
+    int sense_light = 0, sense_dark = 0;
```

Comment:
As it is right now, update_ball_sensor "alternates" functionality based on which time it is called so they need to be outside the scope of the local function. The first time it checks the light emission when the emitter is on, the second it checks it when it is off.

I was mostly worried about a false reading if I tried to do them within the same time step. I can test it later with the mock setup that we had running to see if there is any noticeable problem.